app/code/Magento/CmsGraphQl/Model/Resolver/DataProvider/Block.php

```diff
@@ -27,29 +31,67 @@ class Block
      */
     private $widgetFilter;
 
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
     /**
      * @param BlockRepositoryInterface $blockRepository
      * @param FilterEmulate $widgetFilter
+     * @param StoreManagerInterface $storeManager
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
      */
     public function __construct(
         BlockRepositoryInterface $blockRepository,
-        FilterEmulate $widgetFilter
+        FilterEmulate $widgetFilter,
+        StoreManagerInterface $storeManager,
+        SearchCriteriaBuilder $searchCriteriaBuilder
     ) {
         $this->blockRepository = $blockRepository;
         $this->widgetFilter = $widgetFilter;
+        $this->storeManager = $storeManager;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
     }
 
     /**
      * Get block data
      *
      * @param string $blockIdentifier
      * @return array
-     * @throws NoSuchEntityException
+     * @throws LocalizedException
      */
     public function getData(string $blockIdentifier): array
     {
-        $block = $this->blockRepository->getById($blockIdentifier);
+        $filterBy = BlockInterface::IDENTIFIER;
+        $storeId = (int)$this->storeManager->getStore()->getId();
```

Comment:
Store needs to be passed to this method from resolver `$context->getExtensionAttributes()->getStore()`.